### PR TITLE
virtualservice delegate support sidecar

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Without this, it is impossible to support the below demo:

root vs

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: gateway-vs
spec:
  hosts:
  - "100.94.29.239"
  - productpage
  - nginx
  gateways:
  - gateway.istio-system.svc.cluster.local
  - mesh
  http:
  - match:
    - uri:
        prefix: "/"
    delegate:
        name: nginx-route
        namespace: default

```

delegate:

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: nginx-route
  namespace: default
spec:
  http:
  - match:
    - gateways:
      - gateway-http.istio-system.svc.cluster.local
	  uri:
        prefix: "/productpage"
    - gateways:
      - mesh
      sourceNamespace: "default"
    route:
    - destination:
        host: nginx
        port:
          number: 80
        subset: v1
```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/model/virtualservice.go [hunk 1]
```diff
@@ -364,8 +382,30 @@ func hasConflict(root, leaf *networking.HTTPMatchRequest) bool {
...
 	}
 
-	// sourceLabels, sourceNamespace and gateways do not apply to delegate
-	// and are empty, so no conflict for them.
+	// sourceNamespace
+	if root.SourceNamespace != "" && leaf.SourceNamespace != root.SourceNamespace {
+		return true
+	}
+
+	// sourceLabels
```
Comment (ramaraochavali) [inline pilot/pkg/model/virtualservice.go:390]: Can you please add more comments here and below
- Labels should not be conflicting?
- Root should have super set of gateways?

File: pilot/pkg/model/virtualservice_test.go [hunk 2]
```diff
@@ -1509,6 +1592,36 @@ func TestHasConflict(t *testing.T) {
...
+			root: &networking.HTTPMatchRequest{
+				SourceNamespace: "test1",
+			},
+			leaf: &networking.HTTPMatchRequest{
+				SourceNamespace: "test2",
+			},
+			expected: true,
+		},
+		{
+			name: "sourceNamespace mismatch",
```
Comment (ramaraochavali) [inline pilot/pkg/model/virtualservice_test.go:1616]: nit: gateway incorrect - root has less gateways than delegate

**Review by ramaraochavali — COMMENTED** [review]
lgtm. couple of comments

**Review by ramaraochavali — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@istio/wg-test-and-release-maintainers ptal

**Pull request merged**